### PR TITLE
[Antithesis | Range Queries] Part 3: Interactive Range Query Support

### PR DESCRIPTION
**Pull request by jeremyk-91**
## General
**Before this PR**: A workload server transaction can try and call getRowColumnRange, but it'll just get an UnsupportedOperationException

**After this PR**:

==COMMIT_MSG==
getRowColumnRange is supported in the workload server.
==COMMIT_MSG==

**Priority**: P2

**Concerns / possible downsides (what feedback would you like?)**: Not much

**Is documentation needed?**: No

## Compatibility
Antithesis Workload server change.

## Testing and Correctness
**What, if any, assumptions are made about the current state of the world? If they change over time, how will we find out?**: Nothing in particular.

**What was existing testing like? What have you done to improve it?**: Added a bunch of tests for the new endpoint. Some of it does overlap with the testing of SnapshotTransaction which I don't really like, but think is tolerable to gather confidence that we call the underlying SnapshotTransaction correctly.

**If this PR contains complex concurrent or asynchronous code, is it correct? The onus is on the PR writer to demonstrate this.**: N/A

**If this PR involves acquiring locks or other shared resources, how do we ensure that these are always released?**: N/A

## Execution
Antithesis Workload server change.

## Scale
Antithesis Workload server change.

## Development Process
**Where should we start reviewing?**: AtlasDbInteractiveTransaction, or the Test, depending on your taste.

**If this PR is in excess of 500 lines excluding versions lock-files, why does it not make sense to split it?**: N/A

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@sverma30
@raiju

### CODE REVIEW

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/store/AtlasDbInteractiveTransaction.java [hunk 1]
```diff
@@ -91,6 +100,36 @@ public void delete(String table, WorkloadCell workloadCell) {
                 workloadCell);
     }
 
+    @Override
+    public List<ColumnValue> getRowColumnRange(String table, Integer row, ColumnRangeSelection columnRangeSelection) {
+        return run(
+                tableReference -> {
+                    // Having a non-configurable batch hint is a bit iffy, but suffices as this won't be used in
+                    // production.
+                    Map<byte[], Iterator<Entry<Cell, byte[]>>> iterators = transaction.getRowsColumnRangeIterator(
```
Comment (mdaudali) [inline atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/store/AtlasDbInteractiveTransaction.java:109]: We currently ignore the case where this returns multiple rows, but given that we only request a single row, that would be a bug. We could add a precondition here to surface that bug, if here's the relevant place to do so

File: atlasdb-workload-server-api/src/main/java/com/palantir/atlasdb/workload/transaction/InteractiveTransaction.java [hunk 2]
```diff
@@ -31,5 +32,7 @@ public interface InteractiveTransaction {
 
     void delete(String table, WorkloadCell workloadCell);
 
+    List<ColumnValue> getRowColumnRange(String table, Integer row, ColumnRangeSelection columnRangeSelection);
```
[the remark this answers is not in the document]
Comment (jeremyk-91) [reply atlasdb-workload-server-api/src/main/java/com/palantir/atlasdb/workload/transaction/InteractiveTransaction.java:35]: Is fair. I went for this for consistency with `write`, but that's a separate thing I can check with @Sam-Kramer when he's back.

File: atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/store/AtlasDbInteractiveTransactionTest.java [hunk 3]
```diff
@@ -77,6 +82,95 @@ public void readHandlesEmptyAndPresentValue() {
...
     }
 
+    @Test
+    public void witnessRecordsColumnRangeActions() {
+        assertThat(readWrite(transaction -> {
+                    transaction.write(TABLE_1, WORKLOAD_CELL_TWO, VALUE_ONE);
+                    transaction.getRowColumnRange(
+                            TABLE_1,
+                            WORKLOAD_CELL_TWO.key(),
+                            ColumnRangeSelection.builder().build());
```
[the remark this answers is not in the document]
Comment (jeremyk-91) [reply atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/store/AtlasDbInteractiveTransactionTest.java:92]: 👍  added. It's only frequent in the tests, but seems fine.

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/store/AtlasDbInteractiveTransaction.java (hunk 1 above)
Comment (jeremyk-91) [reply atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/store/AtlasDbInteractiveTransaction.java:109]: Yep, added a precondition for returning just one.

**Pull request merged**